libs/pipelines/api_pipeline.py

```diff
@@ -0,0 +1,106 @@
+from collections import namedtuple
+import logging
+
+from libs.enums import Intervention, AggregateLevel
+from libs import validate_results
+from libs import build_processed_dataset
+from libs import dataset_deployer
+from libs.us_state_abbrev import US_STATE_ABBREV
+
+from libs.datasets import results_schema as rc
+from libs.functions import generate_api as api
+
+logger = logging.getLogger(__name__)
+PROD_BUCKET = "data.covidactnow.org"
+
+APIPipelineProjectionResult = namedtuple(
+    "APIPipelineProjectionResult", ["intervention", "aggregate_level", "projection_df",]
+)
+
+APIPipelineAPIRow = namedtuple("APIPipelineAPIRow", ["key", "api"])
+
+APIPipelineResult = namedtuple("APIPipelineResult", ["api_tuples"])
```

Comment:
How do you feel about these?  I think these classes could be dataclasses to add some type inference.  don't have a great idea what api_tuples contains from just reading this

Reply:
ya i truly was at a loss for how to name these

Reply:
i think classes would bloat them out tbh?

Reply:
normally it's like if you can't think of good names do they need to be named?

Reply:
i renamed them a bit and i'm gonna ack this for the time being